controllers/workspace/devworkspace_controller.go

```diff
@@ -67,6 +67,107 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+// applyHomeInitDefaults applies default values for image and command fields
+// of the init-persistent-home container.
+func applyHomeInitDefaults(c corev1.Container, workspace *common.DevWorkspaceWithConfig) (corev1.Container, error) {
+	if c.Image == "" {
+		inferred := home.InferWorkspaceImage(&workspace.Spec.Template)
+		if inferred == "" {
+			return c, fmt.Errorf("unable to infer workspace image for init-persistent-home; specify image explicitly")
+		}
+		c.Image = inferred
+	}
+	if len(c.Command) == 0 {
+		c.Command = []string{"/bin/sh", "-c"}
+	}
+	return c, nil
+}
+
+// validateNoAdvancedFields validates that the init-persistent-home container
+// does not use advanced Kubernetes container fields that could make behavior unpredictable.
+func validateNoAdvancedFields(c corev1.Container) error {
+	if len(c.Ports) > 0 {
+		return fmt.Errorf("ports are not allowed for init-persistent-home")
+	}
+
+	if c.LivenessProbe != nil || c.ReadinessProbe != nil || c.StartupProbe != nil {
+		return fmt.Errorf("probes are not allowed for init-persistent-home")
+	}
+
+	if c.Lifecycle != nil {
+		return fmt.Errorf("lifecycle hooks are not allowed for init-persistent-home")
+	}
+
+	if c.Stdin || c.StdinOnce || c.TTY {
+		return fmt.Errorf("stdin/tty fields are not allowed for init-persistent-home")
+	}
+
+	if len(c.VolumeDevices) > 0 || c.WorkingDir != "" {
+		return fmt.Errorf("volumeDevices and workingDir are not allowed for init-persistent-home")
```

Comment:
I'm slightly confused. Maybe I don't have the complete context. Why are `VolumeDevices` and `WorkingDir` being validated together here?